src/stashbox.extension.wcf/PerServiceInstanceLifetime.cs

```diff
@@ -8,14 +8,12 @@ namespace Stashbox.Extension.Wcf
 {
     public sealed class PerServiceInstanceLifetime : LifetimeBase
     {
+        private volatile Expression _expression;
```

Comment:
@devworker55 Hi Brandon, it's only a recommendation for double check locking, with the 'volatile' keyword the '_expression' variable will be accessed without cpu optimizations, directly from the memory every time, so always the actually latest value will be in it without any register caching.

There's a lot interesting posts about the pros and contras of this like it's actually unnecessary because the lock statement's memory barrier already ensures this, or it's recommended, because it's actually not needed on existing architectures but who knows how this will behave on future platforms without the 'volatile' and so on. If you are more interested, [this](https://stackoverflow.com/questions/1964731/the-need-for-volatile-modifier-in-double-checked-locking-in-net) could be a good starting point, it could guide you deeper and deeper in this topic :)

Actually i'm now thinking about how could i achieve the same result but without the whole double check locking stuff, but i don't have any result yet.